web/blog/vast-v2.4.1/index.md

```diff
@@ -0,0 +1,61 @@
+---
+title: VAST v2.4.1
+description: Faster Initial Results
+authors: dominiklohmann
+date: 2022-12-19
+tags: [release, feather, performance]
+---
+
+[VAST v2.4.1][github-vast-release] improves the performance of queries when VAST
+is under high load, and significantly reduces the time to first result for
+queries with a low selectivity.
+
+[github-vast-release]: https://github.com/tenzir/vast/releases/tag/v2.4.1
+
+<!--truncate-->
+
+## Reading Feather Files Incrementally
+
+VAST's Feather store naïvely used the [Feather reader][feather-reader] from the
+Apache Arrow C++ library in its initial implementation. However, its API is
+rather limited: It does not support reading incrementally. We've swapped this
+out with a more efficient implementation that does.
+
+[feather-reader]: https://github.com/apache/arrow/blob/apache-arrow-10.0.1/cpp/src/arrow/ipc/feather.h#L57-L108
+
+This is best explained visually:
+
+![Incremental Reads](incremental-reads-light.png#gh-light-mode-only)
+![Incremental Reads](incremental-reads-dark.png#gh-dark-mode-only)
+
+Within the scope of a single store, queries take the same amount of time
+overall, but there exist two distinct advantages of this approach:
+1. The first result arrives much faster at the client.
+2. Stores do less work for cancelled queries.
+
+One additional benefit that is not immediately obvious only comes into play a
+query arrives at multiple stores in parallel. Because the dksk reads are more
+evenly spread out now, the disk reads are less likely to overlap between stores
+queries in parallel. For disk-read performance-bound deployments this can lead
+to a significant query performance improvement.
+
+To verify and test this, I've created a VAST database with 300M Zeek events from
```

Comment:
```suggestion
To verify and test this, we've created a VAST database with 300M Zeek events.
```

Events alone is not sufficient to express the juiciness of a dataset. You need to add the actual size in GiB.

Reply:
I've added this in parantheses.